### ISSUE TITLE
Links, buttons and footer accessibility issues

### ISSUE THREAD

**Issue by kevindherman** [proudcity/wp-proudcity#1780]
@lukefretwell @aschmoe here are some more items from the Digital Marin accessibility team.

Hi Kevin,

I did more accessibility testing with the NVDA screen reader on our new website https://digitalmarin-ca.proudcity.com/. This time I was looking more closely at the links/buttons that are between the main content and the footer: Helpful, Share, Size. Here are the accessibility problems I found:

1. All 3 of these items (which I think ProudCity calls “footer actions” in the Editor) are announced by the screen reader as being in a “Navigation Landmark”. But I would argue that these have nothing to do with navigating this website at all. They are really part of a special kind of tool bar. Looking at the HTML, this section is inside of a <nav> element. Would you consider using a generic landmark role instead of the <nav> element and giving it an aria label of some sort of tool bar instead? 

2. The screen reader announces all 3 of these as links, but they really all function as buttons and not links in spite of them all being coded using the <a> HTML element. Links take you to a new page or to a section on the current page, but these do other things. I just found a really great guide on CSS-Tricks for buttons and links that helps to decide how to choose between them.
 
3. Helpful: The screen reader announces “Helpful link this makes me proud”. Clicking on this link (which is styled as a square button) only changes the color of the button. It is not going to a new page or jumping to a new section of this page.

a. As a screen reader user I would have no clue that anything happened when I activated the link; nothing is announced by the screen reader, so it will be confusing.
b. For the sighted user, it still isn’t clear to me what clicking this button will do. When I click it, it only makes the icon and text unreadable by making the background blue while the text stays dark. Is this a problem with our theme + color choices? Is it something you can fix on your end or do we need special CSS to override the colors used here?
![unnamed (3) - 1](https://user-images.githubusercontent.com/6032426/100131059-88f63e80-2e51-11eb-93b4-b94594109d42.jpg)

4. Share: Clicking on this opens a submenu with 3 actual links (so this does reveal actual navigation for social media sharing). The screen reader announces “Share collapsed submenu link”. This is the one link in this tool bar that makes sense to be a link, but could also be considered a button. It acts more like a button in that both the Enter and the Spacebar activate it.
![image014](https://user-images.githubusercontent.com/6032426/100131087-94496a00-2e51-11eb-9fd9-00b25f69a155.jpg)

5. Size: Clicking on this also opens a submenu. This submenu has 3 items that are announced as links but really function as buttons. The screen reader says “Size collapsed submenu link”, and clicking it shows the submenu and changes the state from collapsed to expanded as it should. Clicking on an item in the submenu changes the visual appearance of some of the text on the page but does not link to some other page or section of this page. But aside from calling them links or buttons, there are other problems with this font resizing widget.
![unnamed (3)](https://user-images.githubusercontent.com/6032426/100131121-9e6b6880-2e51-11eb-90eb-0d1970d2aec0.jpg)

a. Visually, the decrease font size link/button is just a regular “a”, not the Font Awesome “A” used in the first link to increase font size.
b. The superscripted “-“ is just a hyphen, not a minus HTML entity. As such, NVDA completely ignores it (due to my settings to not speak punctuation). I think using the HTML entity “&minus;” should be used here instead, which NVDA will announce as “minus”. (Note that NVDA does already say “plus” for the “+”.) When I explore the popup menu with NVDA using the arrow keys, I just hear “link a” for that last link. If I explore the popup menu with NVDA using the tab key, I hear “a link Decrease Font Size”.

 

Putting these together, I think this would be better code (which is just modeled after the code for the Increase Font Size link without dealing with the link vs. button question):

<li>

    <a href="#" title="Decrease Font Size" class="decreaseFont">

        <i aria-hidden="true" class="fa fa-font"></i>

        <sup>&minus;</sup>

    </a>

</li>

Instead of what is currently coded:

<li>

<a href="#" title="Decrease Font Size" class="decreaseFont">

a

<sup>-</sup>

</a>

</li>

 

NVDA will then say “minus” and not “a” when it encounters this button/link.

**Comment by aschmoe** [comment]
@lukefretwell this will go out with release
